jobs/JGLOBAL_STAGE_IC

```diff
@@ -5,13 +5,99 @@ source "${HOMEgfs}/ush/jjob_header.sh" -e "stage_ic" -c "base stage_ic"
 
 # Restart conditions for GFS cycle come from GDAS
 # shellcheck disable=SC2153
-rCDUMP=${RUN}
+rRUN=${RUN}
 # shellcheck disable=SC2153
-[[ ${RUN} = "gfs" ]] && export rCDUMP="gdas"
-export rCDUMP
+[[ ${RUN} = "gfs" ]] && export rRUN="gdas"
 
-# Execute the Script
-"${SCRgfs}/exglobal_stage_ic.sh"
+# Define significant cycles
+# Locally scoped variables and functions
+# shellcheck disable=SC2153
+half_window=$(( assim_freq / 2 ))
+current_cycle="${PDY}${cyc}"
+previous_cycle=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} - ${assim_freq} hours" +%Y%m%d%H)
+current_cycle_begin=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} - ${half_window} hours" +%Y%m%d%H)
+current_cycle_end=$(date --utc -d "${current_cycle:0:8} ${current_cycle:8:2} + ${half_window} hours" +%Y%m%d%H)
+current_cycle_offset=$(date --utc -d "${PDY} ${cyc} + ${OFFSET_START_HOUR} hours" +%Y%m%d%H)
+
+# Define model start date for current_cycle as the time the forecast will start
+if [[ "${DOIAU:-NO}" == "YES" ]]; then
+  model_start_date_current_cycle="${current_cycle_begin}"
+else
+  if [[ "${REPLAY_ICS:-NO}" == "YES" ]]; then
+    model_start_date_current_cycle=${current_cycle_end}
+  else
+    model_start_date_current_cycle=${current_cycle}
+  fi
+fi
+export previous_cycle current_cycle_offset model_start_date_current_cycle
+
+# Define MEMDIR_ARRAY
+MEMDIR_ARRAY=()
+if [[ "${RUN:-}" = "enkfgdas" || "${RUN:-}" = "gefs" ]]; then
+  if [[ "${RUN:-}" = "gefs" ]]; then
+    ii_start=0
+  elif [[ "${RUN:-}" = "enkfgdas" ]]; then
+    ii_start=1
+  fi
+  # Populate the member_dirs array based on the value of NMEM_ENS
+  for ((ii = "${ii_start}"; ii <= "${NMEM_ENS:-0}"; ii++)); do
+    MEMDIR_ARRAY+=("mem$(printf "%03d" "${ii}")")
+  done
+else
+  MEMDIR_ARRAY+=("")
+fi
+
+# Initialize return code
+err=0
+
+###############################################################
+for MEMDIR in "${MEMDIR_ARRAY[@]}"; do
```

Comment:
Not sure about moving this loop into the j-job. But, this is also a non-production script, so maybe it doesn't matter.

Reply:
I spoke with Rahul about this, gonna leave it in the JJOB for now. The issue is needing to re-declare the COM variables for each member inside the python/yaml/jinja. Can come back and update this in a later PR.

Reply:
Well, the ex-script could've done the same thing and called the python. But I'm fine leaving it.

Reply:
Could you provide an example of calling the `declare_from_tmpl` function from the python ex-script? I attempted and couldn't figure it out.

Reply:
This can actually be done right in the Jinja template.  See `parm/archive/master_enkf.yaml.j2` as an example:
https://github.com/NOAA-EMC/global-workflow/blob/8fee36f0307b0c08080e3a8fa8fa6703e7da5fce/parm/archive/master_enkf.yaml.j2#L52-L71

Reply:
Thanks @DavidHuber-NOAA ! Let me try to incorporate something similar into my staging branch...